RealmSwift/Sync.swift

```diff
@@ -841,66 +823,66 @@ public extension MongoCollection {
         return Future { promise in
             self.updateOneDocument(filter: filter, update: update, upsert: upsert) { updateResult, error in
                 if let updateResult = updateResult {
                     promise(.success(updateResult))
                 } else {
-                    promise(.failure(error ?? App.UserError.uncertainState))
+                    promise(.failure(error ?? NSError(.fail, message: "Promise failed")))
                 }
             }
         }
     }
 
     /// Updates a single document matching the provided filter in this collection.
-    /// - Parameters:
-    ///   - filter: A bson `Document` representing the match criteria.
-    ///   - update: A bson `Document` representing the update to be applied to a matching document.
+    /// @param filter: A bson `Document` representing the match criteria.
+    /// @param update: A bson `Document` representing the update to be applied to a matching document.
+    /// @returns A publisher that eventually return `UpdateResult` or `Error`.
     func updateOneDocument(filter: Document, update: Document) -> Future<UpdateResult, Error> {
         return Future { promise in
             self.updateOneDocument(filter: filter, update: update) { updateResult, error in
                 if let updateResult = updateResult {
                     promise(.success(updateResult))
                 } else {
-                    promise(.failure(error ?? App.UserError.uncertainState))
+                    promise(.failure(error ?? NSError(.fail, message: "Promise failed")))
                 }
             }
         }
     }
 
     /// Updates multiple documents matching the provided filter in this collection.
-    /// - Parameters:
-    ///   - filter: A bson `Document` representing the match criteria.
-    ///   - update: A bson `Document` representing the update to be applied to a matching document.
-    ///   - upsert: When true, creates a new document if no document matches the query.
+    /// @param filter: A bson `Document` representing the match criteria.
+    /// @param update: A bson `Document` representing the update to be applied to a matching document.
+    /// @param upsert: When true, creates a new document if no document matches the query.
+    /// @returns A publisher that eventually return `UpdateResult` or `Error`.
     func updateManyDocuments(filter: Document, update: Document, upsert: Bool) -> Future<UpdateResult, Error> {
         return Future { promise in
             self.updateManyDocuments(filter: filter, update: update, upsert: upsert) { updateResult, error in
                 if let updateResult = updateResult {
                     promise(.success(updateResult))
                 } else {
-                    promise(.failure(error ?? App.UserError.uncertainState))
+                    promise(.failure(error ?? NSError(.fail, message: "Promise failed")))
```

Comment:
Seems like this error is repeatedly returned all over– can we create a static error somewhere or add a case of error somewhere to not have this repeated all over?